### DISCUSSION TITLE
File Transfer using QPSK

### DISCUSSION THREAD

**Discussion by engrfazalkhan** [duggabe/gr-control#14] [Q&A] [answered]
Hi. I have been successful in transitting file using BPSK over the air with SDR. Now I am trying to do the same with QPSK first by using ZMQ and then switching to OTA transmision. However, I am unable to do so. Can you help me out with this?
![qpsk_transmitter](https://github.com/user-attachments/assets/0a405bf9-199a-4e55-b203-cde7d83a6d77)
![qpsk_receiver](https://github.com/user-attachments/assets/a33aba63-ce30-4868-b831-04537ef6f7a4)
![transmit](https://github.com/user-attachments/assets/a5a63efa-4260-4b79-bdfc-0bb02a9efb3c)

![receive](https
![constellation at transmitter](https://github.com/user-attachments/assets/076c004f-1b32-497e-859f-c19d566f4e78)
://github.com/user-attachments/assets/904b9d54-4430-4fae-88ee-63549ada70da)
![receive](https://github.com/user-attachments/assets/87d3d019-2120-4890-acd4-42f686151267)

**Comment by duggabe** [comment, marked as answer]
Since I have not written a "File Transfer using QPSK" package, it is task to be done by the student. If you join the GNU Radio General Chat room #gnuradio:gnuradio.org, you can ask specific questions there. See https://wiki.gnuradio.org/index.php?title=Chat

**Comment by LeonidLd** [comment]
Good afternoon could you please share your working diagram of BPKS ?

**Reply by duggabe** [reply]
Read and follow https://wiki.gnuradio.org/index.php?title=File_transfer_using_Packet_and_BPSK
Be sure to do all of the Prerequisites first!
